docs/src/pages/components/data-grid/selection/ControlledSelectionGrid.tsx

```diff
@@ -10,15 +10,16 @@ export default function ControlledSelectionGrid() {
   });
 
   // TODO: https://github.com/mui-org/material-ui-x/issues/246
-  const [, setSelection] = React.useState<RowId[]>([]);
+  const [selectionModel, setSelectionModel] = React.useState<RowId[]>([]);
 
   return (
     <div style={{ height: 400, width: '100%' }}>
       <DataGrid
         checkboxSelection
         onSelectionChange={(newSelection) => {
-          setSelection(newSelection.rowIds);
+          setSelectionModel(newSelection.selectedRows.map((row) => row.id));
```

Comment:
This signature is weird, to not say inconsistent.

The interface of `onSortModelChange` is:

```tsx
onSortModelChange?: (params: SortModelParams) => void;

interface SortModelParams {
  /**
   * The sort model used to sort the grid.
   */
  sortModel: SortModel;
  /**
   * The full set of columns.
   */
  columns: Columns;
  /**
   * Api that let you manipulate the grid.
   */
  api: any;
}
```

The interface of `onFilterModelChange` is:

```tsx
onFilterModelChange?: (params: FilterModelParams) => void;

interface FilterModelParams {
  /**
   * The filter model.
   */
  filterModel: FilterModel;
  /**
   * The full set of columns.
   */
  columns: Columns;
  /**
   * The full set of rows.
   */
  rows: RowModel[];
  /**
   * Api that let you manipulate the grid.
   */
  api: any;
}
```
```suggestion
          setSelectionModel(params.selectionModel);
```

Reply:
Yes it's a bit inconsistent. What should we provide?
- Api (as it allows to retrieve everything)
- The event model (filterModel, sortModel...)
- ?

Reply:
IMHO, it should have two args, nothing else, nothing more:

(event, selectionModel)

Reply:
I strongly feel that passing the api could be really beneficial specially for ppl that are still on class based component 🤔

Reply:
I see 3 reasons not to expose the apiRef:

- the apiRef is supposed to be XGrid only. It would prevent XGrid to truly be an extension of DataGrid, as we would need to change the signature or expose it as null which is deceptive.
-  Expose the apiRef everywhere wouldn't scale, where would we stop?
- class components can use refs, nothing structurally wrong with it. It's a design limitation with the current useApiRef approach. It's mostly fixed in #990. It's actually another reason to remove useApiRef from the public API (what we started to discuss it as a possible follow-up). Proof: https://codesandbox.io/s/material-demo-forked-wvq55?file=/demo.tsx. It seems that setTimeout is still required because of the two effects cascading hydration of the apiRef (which is probably OK).